src/components/new-safe/create/steps/StatusStep/StatusStepper.tsx

```diff
@@ -1,40 +1,91 @@
 import { Box, Step, StepConnector, Stepper, Typography } from '@mui/material'
+import { useEffect } from 'react'
 import css from '@/components/new-safe/create/steps/StatusStep/styles.module.css'
 import EthHashInfo from '@/components/common/EthHashInfo'
 import { SafeCreationStatus } from '@/components/new-safe/create/steps/StatusStep/useSafeCreation'
 import StatusStep from '@/components/new-safe/create/steps/StatusStep/StatusStep'
 import { usePendingSafe } from './usePendingSafe'
+import { useHasFeature } from '@/hooks/useChains'
+import { useWeb3ReadOnly } from '@/hooks/wallets/web3'
+import { FEATURES } from '@/utils/chains'
+
+// TODO: Remove when we update SDK
+const useSyncZkEvmSafeAddress = (status: SafeCreationStatus) => {
+  const shouldFetchAddress = useHasFeature(FEATURES.SAFE_CREATION_FETCH_ADDRESS)
+  const [pendingSafe, setPendingSafe] = usePendingSafe()
+  const web3ReadOnly = useWeb3ReadOnly()
+
+  useEffect(() => {
+    if (
+      !shouldFetchAddress ||
+      status !== SafeCreationStatus.SUCCESS ||
+      !pendingSafe ||
+      !pendingSafe.txHash ||
+      pendingSafe.safeAddress ||
+      !web3ReadOnly
+    ) {
+      return
+    }
+
+    let isCurrent = true
+
+    web3ReadOnly
+      .getTransactionReceipt(pendingSafe.txHash)
+      .then((receipt) => {
+        if (isCurrent) {
+          const safeAddress = receipt.logs[0].address
```

Comment:
We should check the log's topic and search for the first log with the correct one (ProxyCreation)

If we for instance enable relaying on one of those networks there will also be Events from the Gelato Relayer.
Or if we execute the tx from another Smart Contract wallet there might be events from that wallet as well.

Reply:
I've updated it to parse the logs against the ProxyFactory interface.